scripts/publish/aws-s3.sh

```diff
@@ -59,8 +62,20 @@ fi
 
 FILENAME=$(basename "$ARGV_FILE")
 
+if [ "$ARGV_DATE" == "true" ]; then
+  DATE="$(date +"%Y-%m-%d")"
+  S3_KEY="$ARGV_PRODUCT_NAME/$DATE/$ARGV_VERSION/$FILENAME"
+else
+  S3_KEY="$ARGV_PRODUCT_NAME/$ARGV_VERSION/$FILENAME"
+fi
+
 aws s3api put-object \
   --bucket "$ARGV_BUCKET" \
   --acl public-read \
-  --key "$ARGV_PRODUCT_NAME/$ARGV_VERSION/$FILENAME" \
+  --key "$S3_KEY" \
   --body "$ARGV_FILE"
+
+# Escape plus signs when printing the final URL
+URL="$(echo "https://$ARGV_BUCKET.s3.amazonaws.com/$S3_KEY" | sed 's/\+/%2B/g')"
```

Comment:
Or even something as simple as a cgi-script we can call from this script with `wget` (using a secret CI var for authentication) that updates a static (publicly viewable) html list of snapshot builds that we can view via a web-browser.